axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/DefaultSegmentTransformer.java

```diff
@@ -126,55 +129,82 @@ private void closeTransactionIterator() {
-                            Event replacement = transformedEvent.getEvent();
-                            if (Event.getDefaultInstance().equals(replacement)) {
-                                event = Event.newBuilder()
-                                             .setTimestamp(event.getTimestamp())
-                                             .setAggregateIdentifier(event.getAggregateIdentifier())
-                                             .setAggregateSequenceNumber(event.getAggregateSequenceNumber())
-                                             .setPayload(EMPTY_PAYLOAD)
-                                             .build();
-                            } else {
-                                event = replacement;
-                            }
-                            done = true;
-                        }
-                    } else if (!transactionIteratorRef.get().hasNext()) {
-                        done = true;
+                    Event event = originalTX.getEvent(transformedTransaction.size());
+                    if (transformedEvent.getToken() == originalTX.getToken() + transformedTransaction.size()) {
+                        event = transform(event, transformedEvent.getEvent());
+                        found = true;
                     }
                     transformedTransaction.add(event);
 
-                    if (originalTX.getEvents().size() == transformedTransaction.size()) {
-                        SegmentWriter segmentWriter = segmentWriterRef.get();
-                        segmentWriter.write(transformedTransaction);
-                        originalTransactionRef.set(null);
+                    if (transformedTransaction.size() == originalTX.getEventsCount()) {
+                        write(transformedTransaction);
                         transformedTransaction.clear();
+                        originalTransactionRef.set(null);
                     }
-                } while (!done);
+                } while (!found);
                 sink.success();
             } catch (Exception e) {
                 sink.error(e);
             }
         }).subscribeOn(Schedulers.boundedElastic());
     }
+
+    private Event transform(Event originalEvent, Event replacementEvent) {
+        if (Event.getDefaultInstance().equals(replacementEvent)) {
+            return Event.newBuilder()
+                        .setTimestamp(originalEvent.getTimestamp())
+                        .setAggregateIdentifier(originalEvent.getAggregateIdentifier())
+                        .setAggregateSequenceNumber(originalEvent.getAggregateSequenceNumber())
+                        .setAggregateType(originalEvent.getAggregateType().isEmpty() ? "" : "-")
+                        .setPayload(EMPTY_PAYLOAD)
+                        .build();
+        }
+        return replacementEvent;
+    }
+
+    private SerializedTransactionWithToken originalTransaction() {
```

Comment:
This has a signature that makes you think it's an accessor method. Instead, it changes the state.

Reply:
renamed the operation.